lib/master.js

```diff
@@ -214,7 +225,8 @@ Master.prototype._startWorkers = function(remainingWorkers) {
                     case 'startup_finished':
                         worker.removeListener('exit', workerExit);
                         self._currentStartingWorker = undefined;
-                        resolve(self._startWorkers(--remainingWorkers));
+                        res.push(msg.serviceReturns[0]);
```

Comment:
The worker returns an array of arrays too, for consistency in the `num_workers = 0` case. When we combine those in a single array, we need to unwrap the single element as a result.